src/fmu/dataio/providers/objectdata/_xtgeo.py

```diff
@@ -134,6 +134,21 @@ def export_to_file(self, file: Path | BytesIO) -> None:
 class PolygonsDataProvider(ObjectDataProvider):
     obj: xtgeo.Polygons
 
+    def __post_init__(self) -> None:
+        if self.fmt == FileFormat.csv:
+            self.obj = self.obj.copy()
+
+            self.obj.xname = "X"
+            self.obj.yname = "Y"
+            self.obj.zname = "Z"
```

Comment:
Changed from renaming the dataframe columns just before export, to rename the attributes instead to get everything working properly.

Reply:
https://github.com/equinor/xtgeo/issues/1310